Gems/ROS2/Code/ros2_target_depends.cmake

```diff
@@ -3,29 +3,26 @@
 #
 # SPDX-License-Identifier: Apache-2.0 OR MIT
 
-function(target_depends_on_ros2_packages TARGET_NAME)
-    find_package(ROS2 MODULE)
-    if (NOT ROS2_FOUND)
-        return()
+function(target_depends_on_ros2_package TARGET_NAME )
+    list (GET ARGN 0 _package)
+    find_package(${ARGN})
+    include(${${_package}_DIR}/${_package}Config.cmake OPTIONAL)
+    include(${${_package}_DIR}/${_package}config.cmake OPTIONAL)
+    include(${${_package}_DIR}/${_package}-config.cmake OPTIONAL)
+    include(${${_package}_DIR}/${_package}-Config.cmake OPTIONAL)
+    if( ${${_package}_FOUND} )
+        message(DEBUG "Package ${_package} was found (${${_package}_DIR}) version ${${_package}_VERSION} targets : ${${_package}_TARGETS}")
+        target_link_libraries(${TARGET_NAME} PUBLIC ${${_package}_TARGETS})
```

Comment:
nit: Should add an `else()` clause here to maybe debug print that the package was not found (or is it an error?)

Reply:
I've added error.